### DISCUSSION TITLE
autokey-common could not find packages

### DISCUSSION THREAD

**Discussion by avidseeker** [autokey/autokey#707] [Q&A] [answered]
Tried installing autokey from AUR.
```
yay -S autokey-common
```
gives the error:
```
 -> Could not find all required packages:
        wmctrl (Wanted by: autokey-common)
        xautomation (Wanted by: autokey-common)
        xorg-xwd (Wanted by: autokey-common)
```
am I missing some repos in my mirror list? I tried manually installing these packages (with adding them to PATH), still not solved.

**Comment by josephj11** [comment, marked as answer]
I'm going to answer this in as much detail as I can (not being an Arch user). If you come over to [Gitter](https://gitter.im/autokey/autokey) where our devs and advanced users hang out, you may get better help there. I believe our lead developer runs Arch. We also have Manjaro users.

Let's break this down.

I don't know how Arch packages things, but my understanding is that there are two packages - one that installs AutoKey with the GTK front end and the other that installs the Qt front end. You can't easily have both on Arch for some reason. They both use autokey-common. I know that on Debian-based systems, you have to install at least one front end and the common package at the same time or it gets very upset. It doesn't look like you installed a front end, so you may be using the wrong command.

We have instructions for installing on Arch using pip [here](https://github.com/autokey/autokey/wiki/Installing#arch-linux), but we don't have similar instructions for using aur or another package manager although you'll find some guidance further down in the above link.

Once you get that sorted, there are two sets of dependencies (also detailed in the link above). One set is mandatory and the other is optional.

You're going to need wmctl (at least) for our window-related API calls. I'm not familiar with xorg-xwd, but it is probably also required.

Xautomation is optional. It is used by a few API calls - especially visgrep(), but will not be missed if you don't use those specific calls in your scripts. I've been promoting visgrep() for quite a while, but haven't seen any evidence that it is being actively used.

How to get your paths, etc. magically aligned with AutoKey on Arch is beyond my expertise.
